packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.html

```diff
@@ -4,28 +4,78 @@
+                    class="legend-description-container nui-text-small d-flex flex-column justify-content-center px-2"
+                    [ngClass]="['state-' + chartAssist.renderStatesIndex[legendSeries.id]?.state]">
+
+                        State - {{chartAssist.renderStatesIndex[legendSeries.id]?.state}}
+                    <div *ngIf="isSeriesInteractive(legendSeries); else noLink"
+                        class="legend-description"
+                        (click)="$event.stopPropagation()">
+                        <div *ngIf="!legendSeries.link" 
+                            class="legend-description-primary link" 
+                            [title]="legendSeries.legendDescriptionPrimary"
+                            (click)="onInteraction(legendSeries, $event)">
+                                HANDLER - {{legendSeries.legendDescriptionPrimary}}
+                        </div>
+                        <a class="legend-description-primary link"
+                            *ngIf="legendSeries.link" 
+                            [href]="legendSeries.link" 
+                            rel="noopener noreferrer"
+                            [title]="legendSeries.legendDescriptionPrimary">
+                                HREF - {{legendSeries.legendDescriptionPrimary}}
+                        </a>
+                        <div (click)="onInteraction(legendSeries, $event)"
+                            class="legend-description-secondary"
+                            *ngIf="!legendSeries.secondaryLink" 
+                            [title]="legendSeries.legendDescriptionSecondary">
+                                {{legendSeries.legendDescriptionSecondary}}
+                        </div>
+                        <div *ngIf="legendSeries.secondaryLink">
+                            <a class="legend-description-secondary link"
+                            [href]="legendSeries.secondaryLink" 
+                            rel="noopener noreferrer"
+                            [title]="legendSeries.legendDescriptionSecondary">
+                            HREF - {{legendSeries.legendDescriptionSecondary}}
+                            </a>
+                        </div>
+                    </div>
+                    <ng-template #noLink>
+                        <div class="legend-description">
+                            <div *ngIf="!!legendSeries.legendDescriptionPrimary" 
+                                class="legend-description-primary" 
+                                [title]="legendSeries.legendDescriptionPrimary">{{ legendSeries.legendDescriptionPrimary }}
+
+                            </div>
+                            <div *ngIf="!!legendSeries.legendDescriptionSecondary" class="legend-description-secondary" [title]="legendSeries.legendDescriptionSecondary">{{ legendSeries.legendDescriptionSecondary }}
+
+                            </div>
+                        </div>
+                    </ng-template>
```

Comment:
I wonder if we actually need this extra template here. It seems like it would be possible to use ngClass on the other templates above to conditionally add the `link` class which would provide the correct styling and hover behavior for links only when necessary.

Reply:
If there was a way to conditionally add a click handler to the tag as well I would agree with you. But i don't believe there is?

Reply:
I think you should be able to do something like this:
```html
(click)="isSeriesInteractive ? onClick() : undefined"
```

Reply:
On second thought, it's probably better to go ahead and set the click handler, and just filter out clicks in the handler implementation with an if check:
```ts
public onClick() {
    if (this.isSeriesInteractive) {
        // ...do interactive stuff
    }
}
```